naw/naw.py

```diff
@@ -65,3 +52,54 @@ def main():
             print(elem)
         if re.search(pattern, text, flags=re.IGNORECASE):
             print(text)
+
+
+def set_telephone(data, pattern):
+    for elem in data:
+        name = elem["NAAM"]
+        if re.search(pattern, name, flags=re.IGNORECASE):
+            number = raw_input("Type the new number for {}: \n".format(
+                name)
+            )
+
+            confirm = raw_input(
+                "Change the number for {} to {}? [y/N]:\n".format(
+                    name, number
+                ))
+            if confirm == 'y':
+                change_url = request.urlopen((
+                    'http://buildbot.lizardsystem.nl/cgi-bin/'
+                    'set_telephone?id={}&number={}'.format(
+                        elem['id'],
+                        number
+                    )))
+                print(change_url.read().decode('utf-8'))
+                change_url.close()
+            else:
+                print('Changing nothing for nobody')
+
+
+def main():
+    url_file = request.urlopen(
+        'http://buildbot.lizardsystem.nl/gis/aanwezigheid.json',
+    )
+    json_str = url_file.read().decode('utf-8')
+    url_file.close()
+
+    data = json.loads(json_str)
+
+    try:
+        pattern = sys.argv[1]
```

Comment:
Ik zou het niet "pattern" noemen. Toen ik het zag in de signature van `print_phone_number()` dacht ik "ah, dat is een regex pattern".

Maar het is dus de optie die meegegeven wordt? Hm. `print_phone_number()` gebruikt `pattern niet eens.

En ik zou er argparse even bij pakken voor de netheid.